module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -0,0 +1,281 @@
+use crate::{
+    tx_verifier::{verify_txs_stateless, TransactionAndRawHash},
+    Batch, SequencerOutcome, SlashingReason, TxEffect,
+};
+use borsh::BorshDeserialize;
+use sov_modules_api::{
+    hooks::{ApplyBlobHooks, TxHooks},
+    Context, DispatchCall, Genesis, Hasher, Spec,
+};
+use sov_rollup_interface::{
+    stf::{BatchReceipt, TransactionReceipt},
+    traits::BatchTrait,
+    Buf,
+};
+use sov_state::{StateCheckpoint, WorkingSet};
+use std::{io::Read, marker::PhantomData};
+use tracing::{debug, error};
+
+type Result<T> = std::result::Result<T, ApplyBatchError>;
+
+pub struct AppTemplate<C: Context, RT, Vm> {
+    pub current_storage: C::Storage,
+    pub runtime: RT,
+    pub(crate) checkpoint: Option<StateCheckpoint<C::Storage>>,
+    phantom_vm: PhantomData<Vm>,
+}
+
+pub(crate) enum ApplyBatchError {
```

Comment:
Could you please add public doc comment explaining what kind of hash is in both cases?

Reply:
added